includes/optimizer.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * Safe SVG Optimizer.
+ *
+ * @package safe-svg
+ */
+
+namespace SafeSVG;
+
+use enshrined\svgSanitize\Sanitizer;
+
+if ( ! class_exists( '\SafeSVG\Optimizer' ) ) {
+	/**
+	 * Class \SafeSVG\Optimizer
+	 */
+	class Optimizer {
+		/**
+		 * The name of the nonce to send with the AJAX call.
+		 *
+		 * @var string
+		 */
+		private $nonce_name = 'safe-svg-optimizer';
+		/**
+		 * The class constructor.
+		 */
+		public function __construct() {
+			add_action( 'init', [ $this, 'init' ] );
+		}
+		/**
+		 * Initialize actions.
+		 *
+		 * @return void
+		 */
+		public function init() {
+			if ( true !== $this->is_enabled() ) {
+				return;
+			}
+			add_action( 'admin_enqueue_scripts', [ $this, 'enqueues' ] );
+			add_action( 'wp_ajax_safe_svg_optimize', [ $this, 'optimize' ] );
+		}
+		/**
+		 * Checks if the Optimizer is enabled.
+		 *
+		 * @return bool
+		 */
+		public function is_enabled(): bool {
+			$has_svg_allowed_tags       = has_filter( 'svg_allowed_tags' );
+			$has_svg_allowed_attributes = has_filter( 'svg_allowed_attributes' );
+			/**
+			 * If a dev has added allowed tags or attributes, we should not optimize the SVGs, because the optimizer will not respect their exclusions.
+			 */
+			if ( $has_svg_allowed_tags || $has_svg_allowed_attributes ) {
+				return false;
+			}
+			$params = $this->svgo_params();
+			return ( ! empty( $params ) && is_array( $params ) );
+		}
+		/**
+		 * The SVGO parameters. Developers can use this filter to pass additional parameters or completely disable the optimizer by passing:
+		 * add_filter( 'safe_svg_svgo_params', '__return_false' );
+		 *
+		 * @return mixed|null
+		 */
+		public function svgo_params() {
+			return apply_filters(
+				'safe_svg_svgo_params',
+				[
+					'multipass' => true,
+				]
+			);
+		}
+		/**
+		 * Enqueue the necessary scripts.
+		 *
+		 * @param string $hook The current admin page.
+		 *
+		 * @return void
+		 */
+		public function enqueues( $hook ) {
+			$allowed_hooks = [
+				'options-media.php',
+				'post.php',
+				'post-new.php',
+				'upload.php',
+				'media-new.php',
+			];
+			if ( ! in_array( $hook, $allowed_hooks, true ) ) {
+				return;
+			}
+			wp_enqueue_script(
+				'safe-svg-admin-scripts',
+				SAFE_SVG_PLUGIN_URL . '/dist/safe-svg-admin.js',
+				[ 'wp-data' ],
```

Comment:
```suggestion
				[ 'wp-data', 'utils' ],
```

Just to make sure wpCookies is around. It will be for our purposes but best practice, etc, etc. :)

Reply:
Done!